File: helpdeskspringboot/src/main/java/com/ez/payload/SlaReportDetail.java
package com.ez.payload;

public interface SlaReportDetail {

    // "priority" column = priority id + name + resolved(hours)
    String getPriority();

    // "team" column = team id + name
    String getTeam();

    // "numOfTickets" column.
    // number of tickets between fromDate toDate, and based on [priorityid, teamid].
    long getNumOfTickets();

    // "numOfOntimeTickets" column.
    // number of ontime tickets between fromDate toDate, and based on [priorityid, teamid].
    long getNumOfOntimeTickets();

    // "numOfLatedTickets" column.
    // number of lated tickets between fromDate toDate, and based on [priorityid, teamid].
    long getNumOfLatedTickets();

    // "SlaPercent" column(SLA percentage) = (numOfOntimeTickets/numOfTickets)
    float getSlaPercent();

}

File: helpdeskangular/src/app/payload/SlaReportDetail.ts
export class SlaReportDetail {

    constructor(

        // priority = id + name + resolveIn(hours)
        public priority: string,

        // team = id + name + assignment method
        public team: string,

        // number of tickets from fromDate to toDate of each [priorityid, teamid]
        public numOfTickets: number,

        // number of ontime tickets
        public numOfOntimeTickets: number,

        // number of lated tickets
        public numOfLatedTickets: number,

        // sla percentage
        public slaPercent: string,



    ) {

    }

}
